src/util/bit_writer.rs

```diff
@@ -0,0 +1,439 @@
+/// A marker type that depends on a const bool.
+struct ConstAssert<const N: u8>;
+
+impl<const N: u8> ConstAssert<N> {
+    #[allow(unused)]
+    pub const IN_RANGE: () = assert!(
+        N > 0 && N <= 8,
+        "Bit count must be greater than zero and less than or equal to 8"
+    );
+}
+
+pub trait AsU8 {
+    fn as_u8(&self) -> u8;
+}
+
+impl AsU8 for u8 {
+    fn as_u8(&self) -> u8 {
+        *self
+    }
+}
+
+impl AsU8 for &u8 {
+    fn as_u8(&self) -> u8 {
+        **self
+    }
+}
+
+/// A bit-level writer that allows writing an arbitrary number of bits to a byte buffer.
+///
+/// BitWriter provides precise control over bit-level writes, allowing you to write
+/// between 1 and 8 bits at a time. It handles byte boundaries automatically and
+/// maintains proper alignment of bits within bytes.
+///
+/// * Byte Ordering:
+///     Within each byte, bits are written from most significant (left) to least significant (right).
+///     For example, writing 3 bits with value `0b101` to an empty buffer will result in `0b10100000`.
+pub struct BitWriter<'a> {
+    buf: &'a mut [u8],
+    index: usize,
+    bit_index: u8,
+}
```

Comment:
We get an unfortunate asymmetry between reading and writing VLA where the reading does not have this kind of helper.

Spontaneously it looks a bit over engineered to require ~400 lines of source file to write some bits in a (for the time being) fringe RTP header extension, in a direction that is not a super common use case (since I'm assuming receiving simulcast is more common than sending).

I'm not saying we definitely shouldn't commit this, but I think we need to reason about the future of it and whether it solves a problem hard enough to not be a few helper methods.

Reply:
Np! I was kinda playing with compile time assertions to prevent runtime checks. My original version was in situ so I'll just swap it out

Reply:
Done